### PR TITLE
feat: add support for Quarkus security in UI Unit tests

### PR DESCRIPTION
**Pull request by mcollovati**
## Description

This change allows to write UI Unit tests that profits of Quarkus security testing annotation that can be applied to test methods and test classes to control the security context that the test is run with for mock user.

Fixes vaadin/quarkus#141
Part of #1655 

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Comment by mcollovati** [comment]
This PR currently targets `issues/quarkus-140_quarkus_ui_unit_test` branch because it requires #1735.
Target must be changed after the other PR gets merged.

**Comment by mshabarov** [comment]
This can be now rebased to `main`.

File: vaadin-testbench-unit-quarkus/src/test/java/com/vaadin/testbench/unit/quarkus/SecurityTestConfig.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import io.quarkus.arc.profile.IfBuildProfile;
+import io.quarkus.test.junit.QuarkusTestProfile;
+
+import com.vaadin.flow.router.RouteConfiguration;
+import com.vaadin.flow.server.ServiceInitEvent;
+import com.vaadin.flow.server.auth.NavigationAccessControl;
+import com.vaadin.flow.server.auth.ViewAccessChecker;
+import com.vaadin.testbench.unit.quarkus.mocks.MockQuarkusServletService;
+
+// Empty configuration class used only to be able to bootstrap spring
```
Comment (mshabarov) [inline vaadin-testbench-unit-quarkus/src/test/java/com/vaadin/testbench/unit/quarkus/SecurityTestConfig.java:26]: nit: should mention Quarkus probably

File: vaadin-testbench-unit-quarkus/src/test/java/com/vaadin/testbench/unit/quarkus/QuarkusUnitSecurityViewAccessCheckerTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+@TestProfile(SecurityTestConfig.ViewAccessCheckerConfig.class)
+class QuarkusUnitSecurityViewAccessCheckerTest extends QuarkusUIUnitTest {
+
+    @Test
+    @TestSecurity(user = "john", roles = { "DEV", "PO" })
+    void withMockUser_loggedUser_authenticationInformationAvailableOnRequest() {
+        VaadinRequest request = VaadinRequest.getCurrent();
+        Principal principal = request.getUserPrincipal();
+        Assertions.assertNotNull(principal,
+                "Principal should be provided by Spring Security, but was not found");
```
Comment (mshabarov) [inline vaadin-testbench-unit-quarkus/src/test/java/com/vaadin/testbench/unit/quarkus/QuarkusUnitSecurityViewAccessCheckerTest.java:35]:
```suggestion
                "Principal should be provided by Quarkus Security, but was not found");
```

**Review by mshabarov — CHANGES_REQUESTED** [review]
Couple of nits, otherwise looks good!

File: vaadin-testbench-unit-quarkus/src/test/java/com/vaadin/testbench/unit/quarkus/SecurityTestConfig.java (hunk 1 above)
Comment (mcollovati) [reply vaadin-testbench-unit-quarkus/src/test/java/com/vaadin/testbench/unit/quarkus/SecurityTestConfig.java:26]: Can be completely removed. It's a copy/paste from Spring add-on test

File: vaadin-testbench-unit-quarkus/src/test/java/com/vaadin/testbench/unit/quarkus/SecurityTestConfig.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+// ApplicationContext
+//
```
Comment (mcollovati) [inline vaadin-testbench-unit-quarkus/src/test/java/com/vaadin/testbench/unit/quarkus/SecurityTestConfig.java:28]:
```suggestion
```

File: vaadin-testbench-unit-quarkus/src/test/java/com/vaadin/testbench/unit/quarkus/SecurityTestConfig.java (hunk 1 above)
Comment (mcollovati) [reply vaadin-testbench-unit-quarkus/src/test/java/com/vaadin/testbench/unit/quarkus/SecurityTestConfig.java]: removed

**Review by mshabarov — APPROVED** [review]

**Pull request merged**